src/lib/support/ZclString.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ZclString.h"
+
+namespace chip {
+
+constexpr size_t kBufferMaximumSize = 254;
```

Comment:
a comment sould be useful here: length prefixed data is  using the first byte for length, hence the limitation is we only support sizes of 0-255.

This comes back on why 254 though: Why can't I have 255-length strings? Instead of 255/254 could we use std limits or similar constants?

Reply:
If I'm not mistaken:
- 1 space for the null termination
- 1 space for the length
- uint8_t has space for 256 characters
--------
 Leaves us with space of 254. 

Calculating the maximum using the std numeric_limits now.

Reply:
@andy31415 @lazarkov The reason it's 254 is that in ZCL a length of 255 means something like "no string value available at all".  So if you allow a string with length 255 here, other parts of the system will treat it as either an invalid value or an empty string, depending....

Reply:
https://github.com/project-chip/connectedhomeip/pull/9486 to fix that.